blockchain/beacon/src/lib.rs

```diff
@@ -6,6 +6,38 @@ mod drand;
 mod drand_api;
 mod group;
 mod identity;
+mod mock_beacon;
 
 pub use beacon_entries::*;
 pub use drand::*;
+pub use mock_beacon::*;
+
+use clock::ChainEpoch;
+use std::error::Error;
+
+pub async fn beacon_entries_for_block<B: Beacon>(
+    beacon: B,
+    round: ChainEpoch,
+    prev: &BeaconEntry,
+) -> Result<Vec<BeaconEntry>, Box<dyn Error>> {
+    let max_round = beacon.max_beacon_round_for_epoch(round, prev);
+    if max_round == prev.round() {
+        return Ok(vec![]);
+    }
+    // TODO: this is a sketchy way to handle the genesis block not having a beacon entry
+    let prev_round = if prev.round() == 0 {
+        max_round - 1
```

Comment:
what are you planning on doing with this? seems like you can't rely on max_round not being 0 and causing a panic

Reply:
max_round cant be 0 is prev.round() == 0. That necessarily means that were at least on round >= 1.

Reply:
Ohh I see, this is handling the logic that was done on construction in go impl